src/Arcus.Templates.ServiceBus.Queue/Program.cs

```diff
@@ -82,22 +85,28 @@ public static IHostBuilder CreateHostBuilder(string[] args)
         }
 #if Serilog
         
-        private static void UpdateLoggerConfiguration(
-            HostBuilderContext hostContext,
-            LoggerConfiguration config)
+        private static async Task ConfigureSerilogAsync(IHost host)
         {
-            config.MinimumLevel.Debug()
-                  .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
-                  .Enrich.FromLogContext()
-                  .Enrich.WithVersion()
-                  .Enrich.WithComponentName("Service Bus Queue Worker")
-                  .WriteTo.Console();
+            var secretProvider = host.Services.GetRequiredService<ISecretProvider>();
+            string connectionString = await secretProvider.GetRawSecretAsync(ApplicationInsightsConnectionStringKeyName);
             
-            var instrumentationKey = hostContext.Configuration.GetValue<string>(ApplicationInsightsInstrumentationKeyName);
-            if (!string.IsNullOrWhiteSpace(instrumentationKey))
+            var reloadLogger = (ReloadableLogger) Log.Logger;
+            reloadLogger.Reload(config =>
```

Comment:
We need to reload the Serilog configuration, otherwise we're getting into an endless loop. See: the Arcus secret store requires logging, but the logging requires the secret store. Therefore, we can fix this by first build the `IHost` and afterwards reload the Serilog configuration with the actual configuration.